README.md

```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
+equilateral. All four sides are the same length.
+
+So a square *is a* rectangle, but a rectangle *is not a* square. In
+object-oriented programming, we would consider this *is-a* relationship
+a place to use inheritance. That is, we could define a rectangle, then
+create a square out of that rectangle and add the additional properties.
+The inheritance allows us to reuse the rectangle's properties without
+having to redefine them on the square.
+
+Let's see this in action, first without inheritance:
+
+```js
+// Rectangle constructor
+function Rectangle(sides, width, height) {
+  this.sides = sides;
+  this.width = width;
+  this.height = height;
+  this.area = function() {
+    return this.width * this.height;
+  }
+  this.perimeter = function() {
+    return (this.width + this.height) * 2;
+  }
+}
+
+// Square constructor
+function Square(sides, length) {
+  this.sides = sides;
+  this.width = length;
+  this.height = length;
+  this.area = function() {
+    return this.width * this.height;
+  },
+  this.perimeter = function() {
+    return (this.width + this.height) * 2;
+  }
+}
+
+var rect = new Rectangle(4, 3, 5);
+var square = new Square(4, 2);
+
+rect.area();
+square.area();
+```
+
+This works great, but we've created two objects that have basically the same properties and behaviors.
+The only difference is that `width` and `height` are the same in a
+`square`, so we only take one `length` argument and use it to set
+`width` and `height` to the same thing.
+
+Inheritance allows us to create a square from a rectangle and then alter
+the square as necessary to make a new type of object, without having to
+repeat everything we did when constructing the rectangle. So how can we
+implement this using inheritance in JavaScript?
+
+### Constructor-Based Inheritance
+
+It isn't much of a stretch to see a simple way to make `Square`
+"inherit" from `Rectangle` by just using the constructor and a `call` to
+set `this` to a new square:
+
+```js
+// Rectangle constructor
+function Rectangle(sides, width, height) {
+  this.sides = sides;
+  this.width = width;
+  this.height = height;
+  this.area = function() {
+    return this.width * this.height;
+  }
+  this.perimeter = function() {
+    return (this.width + this.height) * 2;
+  }
+}
+
+// Square constructor
+function Square(sides, length) {
+  Rectangle.call(this, sides, length, length);
+}
+
+var rect = new Rectangle(4, 3, 5);
+var square = new Square(4, 2);
+
+rect.area();
+square.area();
+```
+
+This will work, and it certainly reduces the repeated code. But under
+the hood, there's a problem. Our `square` isn't truly inheriting from
+the prototypical `Rectangle`. It is, in essence, a standalone object
+that only copied what `Rectangle` had available at the time of
+instantiation.
+
+To illustrate, try adding this code:
+
+```js
+Rectangle.prototype.internalAngles = 90;
+rect.internalAngles;
+square.internalAngles;
+```
+
+The `rect` can access `internalAngles`, but `square` cannot. Why? The
```

Comment:
This is really solid, but it's a little hard to follow. Might be important to really hold students' hands through this bit.